### ISSUE TITLE
#deactivate does not work properly for Account

### ISSUE THREAD

**Issue by ntippie** [minimul/qbo_api#47]
If you try to deactivate an account, it will raise the following error:

```ruby
QboApi::BadRequest: 
[{:fault_type=>"ValidationFault", 
:error_code=>"2020", 
:error_message=>"Required param missing, need to supply the required value for the API", 
:error_detail=>"Required parameter Name is missing in the request"}]
```

How do you think it would best be fixed? Using the full object (rather than sparse) for the update seems like an easy solution, but that may open up a number of other issues since there are many name list entities, and some fields act uniquely around updates (and sometimes their behavior changes according to settings, like Payroll being enabled). We could just change `#set_update` to include Name or DisplayName if they are present in the read object.

**Comment by ntippie** [comment]
I have a fix [here](https://github.com/minimul/qbo_api/compare/master...ntippie:master) that I'll PR if you like.

**Comment by minimul** [comment]
I'd rather not disrupt `set_update` too much, although, like you said you need access to the response. 

Here is my [take]( https://github.com/minimul/qbo_api/tree/deact-account).

Perhaps it is too limiting by only sniffing for and fixing `Account`? 

Should we do it case by case or wholesale?

And don't ask me why I didn't use more instance variables, especially for `entity`!

**Comment by ntippie** [comment]
Sorry, I neglected to consider that `#set_update` was used elsewhere! Definitely an issue with overwriting Name for `#update`.  😱

My main concern with your version is that such a generic instance variable is used only for such a specific purpose. It doesn't seem consistent with the rest of the code (which I have no problem with not using many instance variables).

I also considered that in very specific circumstances it would not be threadsafe. It could be an issue with doing parallel requests with Faraday, but if you're doing parallel requests you're probably using `#batch` instead.

I don't think limiting to Account is a problem. I checked Item (the other troublesome name list entity) and Class and they are okay.

What if we just make a `#set_deactivate` that handles Account specifically? It would be private like `#set_update` so perhaps that is the smallest footprint. I also realized my solution within `#set_update` would need to check for both the symbol and string to avoid overwriting (*puts fingers in ears* "what sym/str issue?") but that wouldn't be the case with `#set_deactivate` since there is no payload for `#deactivate`.

**Comment by minimul** [comment]
True on the consistency criticism.

Agreed on `set_deactivate`. I'll have another iteration (or be my guest if you want gin it up) in the AM.

**Comment by minimul** [comment]
Another [iteration](https://github.com/minimul/qbo_api/tree/deact-account).

**Comment by ntippie** [comment]
Looks great! Thanks.

**Comment by minimul** [comment]
Closed #49
